sunbeam-python/sunbeam/commands/manifest.py

```diff
@@ -89,3 +93,60 @@ def show(ctx: click.Context, id: str) -> None:
         click.echo("Error: Not able to connect to Cluster DB")
     except ManifestItemNotFoundException:
         click.echo(f"Error: No manifest exists with id {id}")
+
+
+@click.command()
+@click.option(
+    "-f",
+    "--manifest-file",
+    help="Output file for manifest, defaults to $HOME/.config/openstack/manifest.yaml",
+    type=click.Path(dir_okay=False, path_type=Path),
+)
+@click.pass_context
+def generate(
+    ctx: click.Context,
+    manifest_file: Path | None = None,
+):
+    """Generate manifest file.
+
+    Generate manifest file with the deployed configuration.
+    If the cluster is not bootstrapped, fallback to default
+    configuration.
+    """
+    client: Client = ctx.obj
+
+    if not manifest_file:
+        home = os.environ.get("SNAP_REAL_HOME")
+        manifest_file = Path(home) / ".config" / "openstack" / "manifest.yaml"
+
+    LOG.debug(f"Creating {manifest_file} parent directory if it does not exist")
+    manifest_file.parent.mkdir(mode=0o775, parents=True, exist_ok=True)
+
+    try:
+        preflight_checks = [DaemonGroupCheck(), VerifyBootstrappedCheck(client)]
+        run_preflight_checks(preflight_checks, console)
+        manifest_obj = Manifest.load_latest_from_clusterdb(
+            client, include_defaults=True
+        )
+    except Exception as e:
+        LOG.debug(e)
+        LOG.debug("Fallback to generating manifest with defaults")
+        manifest_obj = Manifest.get_default_manifest(client)
+
+    try:
+        manifest_dict = asdict_with_extra_fields(manifest_obj)
+        LOG.debug(f"Manifest dict with extra fields: {manifest_dict}")
+        manifest_yaml = yaml.safe_dump(manifest_dict, sort_keys=False)
+
+        # add comment to each line
+        manifest_lines = [f"#{line}" for line in manifest_yaml.split("\n")]
+        manifest_yaml_commented = "\n".join(manifest_lines)
```

Comment:
```suggestion
        manifest_lines = (f"# {line}" for line in manifest_yaml.split("\n"))
        manifest_yaml_commented = "\n".join(manifest_lines)
```

It's better presented (and supported) by IDEs to have a space after the `#`.

I take advantage of this to nitpick on the list comprehension, this can just be a generator

Reply:
Done